batterymanagement.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include "test.h"
+
+enum languages{English=1,German=2};
+enum languages l;
+
+int TempCheck(float temperature)
+{
+  if(l==1)
+  printf("Temperature ");
+  if(l==2)
+  printf("Temperatur ");
+  int isTemperatureInvalid = checkvaluebreach(temperature,0,45);
+  return isTemperatureInvalid;
+}
+  
+int SocCheck(float soc)
+{
+  if(l==1)
+  printf("State of Charge ");
+  if(l==2)
+  printf("Ladezustand ");
+  int isSocInvalid = checkvaluebreach(soc,20,80);
+  return isSocInvalid;
+}
+  
+int ChargerateCheck(float chargeRate)
```

Comment:
These `...Check` functions are not fully tested - in the sense that they could skip a print, or return only ok/not-ok (no warning facility).
How do you know that the warning facility would work?